### ISSUE TITLE
Remove "static type warning" from test.dart

### ISSUE THREAD

**Issue by munificent** [dart-lang/sdk#30529]
In Dart 2.0, there are no static type *warnings*. Type errors are compile-time errors like any other. Once all tests are migrated over, we can remove "static type warning" as a multitest outcome and "staticWarning" and "missingStaticWarning" as test expectations.

**Comment by srawlins** [comment]
I see that `missingStaticWarning` is gone, but `staticWarning` still exists in `pkg/status_file` and `pkg/test_runner`.

**Comment by munificent** [comment]
Yeah, there are still some multitest sections that use it, I think. I believe it's mostly synonymous with `compile-time error`, but I haven't spent the time to root it out.
